### PR TITLE
Skip typecast when trying to evaluate pointer comparisons

### PR DESCRIPTION
**Pull request by romainbrenguier**
This fix the case where the pointer comparison contains typecasts

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [na] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [na] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [na] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/goto-symex/symex_goto.cpp [hunk 1]
```diff
@@ -89,9 +89,10 @@ static optionalt<renamedt<exprt, L2>> try_evaluate_pointer_comparison(
 {
   const constant_exprt *constant_expr =
```
Comment (smowton) [inline src/goto-symex/symex_goto.cpp:90]: Commit message: concidered -> considered

File: unit/goto-symex/try_evaluate_pointer_comparisons.cpp [hunk 2]
```diff
@@ -78,6 +78,26 @@ SCENARIO(
       }
     }
```
Comment (smowton) [inline unit/goto-symex/try_evaluate_pointer_comparisons.cpp:79]: Commit message: carefuly -> carefully

File: jbmc/regression/jbmc/enum/Enum1.java [hunk 3]
```diff
@@ -1,16 +1,16 @@
-public enum enum1
+public enum Enum1
 {
   VAL1, VAL2, VAL3, VAL4, VAL5;
```
Comment (smowton) [inline jbmc/regression/jbmc/enum/Enum1.java:3]: Commit message: too -> to

**Review by smowton — APPROVED** [review]
Just English stuff

**Comment by tautschnig** [comment]
@romainbrenguier The regression tests don't seem to be happy with the change.

**Review by owen-mc-diffblue — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 3eb067a3808190a54f46024fcb666841bce1909e).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/118456839

**Pull request merged**